protocol/stan/v2/context.go

```diff
@@ -0,0 +1,49 @@
+package stan
+
+import (
+	"context"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/nats-io/stan.go"
+)
+
+type msgKeyType struct{}
+
+var msgKey msgKeyType
+
+// MetadataContextDecorator returns an inbound context decorator which adds STAN message metadata to
+// the current context. If the inbound message is not a *stan.Message then this decorator is a no-op.
+func MetadataContextDecorator() func(context.Context, binding.Message) context.Context {
+	return func(ctx context.Context, m binding.Message) context.Context {
+		if msg, ok := m.(*Message); ok {
+			// we store the raw *stan.Msg on the context, but under a unexported key so only the below accessor
+			// functions can be used to extract this info.
+			return context.WithValue(ctx, msgKey, msg.Msg)
+		}
+
+		return ctx
+	}
+}
+
+func SequenceFrom(ctx context.Context) (uint64, bool) {
```

Comment:
Can you add a godoc to this and below methods?

Reply:
done, I've actually changed the behaviour to keep all the data in a single struct, seems unnecessary to have multiple functions to extract this data because you would have ended up with multiple `if value, ok := FooFrom(ctx); ok { ... }` statements if you wanted to view all of the properties.